src/connector/src/sink/remote.rs

```diff
@@ -0,0 +1,574 @@
+        None => Value::Null,
+        Some(ScalarRefImpl::Int32(v)) => Value::from(v),
+        Some(ScalarRefImpl::Int64(v)) => Value::from(v),
+        Some(ScalarRefImpl::Float32(v)) => Value::from(v.into_inner()),
+        Some(ScalarRefImpl::Float64(v)) => Value::from(v.into_inner()),
+        Some(ScalarRefImpl::Decimal(v)) => Number(v.to_string().parse().unwrap()),
+        Some(ScalarRefImpl::Utf8(v)) => Value::from(v),
+        Some(ScalarRefImpl::Bool(v)) => Value::from(v),
+        Some(ScalarRefImpl::NaiveDate(v)) => Value::from(v.to_string()),
+        Some(ScalarRefImpl::NaiveTime(v)) => Value::from(v.to_string()),
+        Some(ScalarRefImpl::Interval(v)) => Value::from(v.to_string()),
+        Some(ScalarRefImpl::Struct(v)) => Value::from(
+            v.fields_ref()
+                .iter()
+                .map(|v| parse_datum(*v))
+                .collect::<Vec<_>>(),
+        ),
+        Some(ScalarRefImpl::List(v)) => Value::from(
+            v.values_ref()
+                .iter()
+                .map(|v| parse_datum(*v))
+                .collect::<Vec<_>>(),
+        ),
+        _ => unimplemented!(),
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use std::collections::HashMap;
+    use std::sync::Arc;
+    use std::time::Duration;
+
+    use risingwave_common::array;
+    use risingwave_common::array::column::Column;
+    use risingwave_common::array::{ArrayImpl, I32Array, Op, StreamChunk, Utf8Array};
+    use risingwave_common::catalog::{Field, Schema};
+    use risingwave_common::types::DataType;
+    use risingwave_pb::connector_service::sink_response::{
+        Response, StartEpochResponse, SyncResponse, WriteResponse,
+    };
+    use risingwave_pb::connector_service::sink_task::write_batch::Payload;
+    use risingwave_pb::connector_service::sink_task::Request;
+    use risingwave_pb::connector_service::{SinkResponse, SinkTask};
+    use risingwave_pb::data;
+    use tokio::sync::mpsc::{UnboundedReceiver, UnboundedSender};
+
+    use crate::sink::remote::{ConnectorClientImpl, RemoteSink, RemoteSinkResponseImpl};
+    use crate::sink::Sink;
+
+    fn create_mock_sink() -> (
+        RemoteSink,
+        UnboundedSender<SinkResponse>,
+        UnboundedReceiver<SinkTask>,
+    ) {
+        let properties = HashMap::from([("output_path".to_string(), "/tmp/rw".to_string())]);
+
+        let _schema = Schema::new(vec![
+            Field {
+                data_type: DataType::Int32,
+                name: "id".into(),
+                sub_fields: vec![],
+                type_name: "".into(),
+            },
+            Field {
+                data_type: DataType::Varchar,
+                name: "name".into(),
+                sub_fields: vec![],
+                type_name: "".into(),
+            },
+        ]);
+
+        let (request_sender, request_recv) = tokio::sync::mpsc::unbounded_channel();
+        let (resp_sender, resp_recv) = tokio::sync::mpsc::unbounded_channel();
+
+        (
+            RemoteSink {
```

Comment:
Better to add a new method named `for_test` instead of creating the raw struct here.

Reply:
Do you mean a new for-test constructor in `impl RemoteSink`?